src/entity/Post.ts

```diff
@@ -391,6 +417,24 @@ const addPostAndKeywordsToDb = async (
   return data.id;
 };
 
+export const rejectPost = async (
+  con: Connection,
+  data: RejectPostData,
+): Promise<RejectPostResult> => {
+  if (!data || !data?.submissionId) {
+    return { status: 'failed', reason: 'missing submission id' };
+  }
+
+  return con.transaction(async (entityManager) => {
+    try {
+      const submissionId = await rejectSubmission(entityManager, data);
+      return { status: 'ok', submissionId };
+    } catch (error) {
+      throw error;
+    }
+  });
+};
```

Comment:
I think this can go to the private routes if we use it only there.
`addNewPost` is here for compatibility reasons